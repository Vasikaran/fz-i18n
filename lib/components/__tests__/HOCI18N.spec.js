'use strict';

var _HOCI18N = require('..//HOCI18N');

var _HOCI18N2 = _interopRequireDefault(_HOCI18N);

var _I18NProvider = require('../I18NProvider');

var _I18NProvider2 = _interopRequireDefault(_I18NProvider);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTestRenderer = require('react-test-renderer');

var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Test = function Test(props) {
  return _react2.default.createElement(
    'div',
    null,
    'test',
    props.placeHolder
  );
};

var defaultProps = {
  i18NKey: "key1<b>vimal</b>"
};

describe('I18N component', function () {

  it("Should display i18n value", function () {
    Test = (0, _HOCI18N2.default)(["placeHolder"])(Test);
    var ele = _reactTestRenderer2.default.create(_react2.default.createElement(
      _I18NProvider2.default,
      { i18n: { key1: "vimal1" } },
      _react2.default.createElement(Test, { placeHolder: 'key1' })
    ));
    var tree = ele.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should display i18n key", function () {
    Test = (0, _HOCI18N2.default)(["placeHolder"])(Test);
    var ele = _reactTestRenderer2.default.create(_react2.default.createElement(Test, { placeHolder: 'key1' }));
    var tree = ele.toJSON();
    expect(tree).toMatchSnapshot();
  });
});