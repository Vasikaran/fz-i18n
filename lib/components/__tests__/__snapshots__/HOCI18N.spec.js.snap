exports[`I18N component Should display i18n key 1`] = `
<div>
  test
  key1
</div>
`;

exports[`I18N component Should display i18n value 1`] = `
<div>
  test
  vimal1
</div>
`;
